### PR TITLE
Add has key

### PR DESCRIPTION
**Pull request by seongreen**

### CODE REVIEW

File: lib/be_protected/response/verification/rules.rb [hunk 1]
```diff
@@ -22,6 +22,16 @@ def passed?
...
 
+        def has_action?(action)
+          attributes.keys.each do |account|
+            attributes[account].keys.each do |alias_name|
+              return true if attributes[account][alias_name].values.include?(action)
+            end
+          end
+          false
+        end
+
```
Comment (avsh) [inline lib/be_protected/response/verification/rules.rb]: remove line

File: lib/be_protected/response/verification/rules.rb [hunk 2]
```diff
@@ -22,6 +22,16 @@ def passed?
           true
         end
 
+        def has_action?(action)
```
Comment (avsh) [inline lib/be_protected/response/verification/rules.rb]: add spec

**Comment by seongreen** [comment]
🆙

File: spec/be_protected/verification_spec.rb [hunk 3]
```diff
@@ -44,6 +44,23 @@
...
 
+      context "when has_action skip_3ds" do
+        let(:response) {
+          { rules: {
+              'system account' => {'alias 7' => {'Transaction amount more than 90 USD'  => 'skip_3ds'}}  
+            }
+          }
+        }
+
+        it "return true on has_action? call" do
```
Comment (avsh) [inline spec/be_protected/verification_spec.rb]: return`S`

**Comment by seongreen** [comment]
🆙

File: spec/be_protected/verification_spec.rb [hunk 4]
```diff
@@ -44,6 +44,23 @@
 
       it_behaves_like "successful response"
 
+      context "when has_action skip_3ds" do
+        let(:response) {
```
Comment (ba1ash) [inline spec/be_protected/verification_spec.rb]: `do...end` for multiline block, please

File: lib/be_protected/response/verification.rb [hunk 5]
```diff
@@ -13,6 +13,10 @@ def error?
         verifications.any? && verifications.values.any?(&:error?)
       end
 
+      def has_action?(action)
+        !rules.nil? && rules.has_action?(action)
```
Comment (ba1ash) [inline lib/be_protected/response/verification.rb]: `!rules.nil?` looks bad, in my opinion.
Maybe, use simple `rules && rules.has_action?(action)`?

File: spec/be_protected/verification_spec.rb (hunk 4 above)
Comment (ba1ash) [inline spec/be_protected/verification_spec.rb]: And improve hash style, please

File: lib/be_protected/response/verification/rules.rb [hunk 6]
```diff
@@ -22,14 +22,22 @@ def passed?
...
+          attributes.keys.each do |account|
```
Comment (ba1ash) [inline lib/be_protected/response/verification/rules.rb]: I suppose this piece can be written in more simple way:
```Ruby
attributes.values.each do |account_bucket|
  account_bucket.values.each do |alias_name_bucket|
    return true if alias_name_bucket.values.include?(action)
  end
end
```
p.s.: I don't know the domain, therefore, I've picked not enough reasonable names for hash values. Please, change names if you can.

**Comment by seongreen** [comment]
🆙

**Pull request merged**